lib/langchain/llm/ollama.rb

```diff
@@ -132,29 +131,19 @@ def complete(
       }
 
       parameters[:options] = llm_parameters.compact
-
-      response = ""
+      responses_stream = []
 
       client.post("api/generate") do |req|
         req.body = parameters
 
-        req.options.on_data = proc do |chunk, size|
-          chunk.split("\n").each do |line_chunk|
-            json_chunk = begin
-              JSON.parse(line_chunk)
-            # In some instance the chunk exceeds the buffer size and the JSON parser fails
-            rescue JSON::ParserError
-              nil
-            end
-
-            response += json_chunk.dig("response") unless json_chunk.blank?
-          end
+        req.options.on_data = json_responses_chunk_handler do |parsed_chunk|
+          responses_stream << parsed_chunk
 
-          yield json_chunk, size if block
+          yield Langchain::LLM::OllamaResponse.new(parsed_chunk, model: parameters[:model]) if block_given?
```

Comment:
Maybe we start referencing these classes directly here since it's all under the same namespace: `OllamaResponse.new ...`. It's cleaner that way, right?

Reply:
yep. it'll probably work too.

Reply:
updated